Sources/Rownd/framework/Automations/PageEvaluator.swift

```diff
@@ -0,0 +1,195 @@
+        return nil
+    }
+    
+    guard let releaseVersion = getReleaseVersionNumber() else {
+        return nil
+    }
+    
+    guard let releaseVersionNumber = Double(releaseVersion) else {
+        return nil
+    }
+    
+    var selectedCapture: MobileAppPageCapture? = nil
+    for capture in captures {
+        if (capture.platform != "ios" ) {
+            continue
+        }
+        
+        guard let appVersion = Double(capture.capturedOnAppVersion) else {
+            continue
+        }
+        
+        // Don't use a capture version that is greater than the current build version
+        if (appVersion > releaseVersionNumber) {
+            continue
+        }
+        
+        guard let newCapture = selectedCapture else {
+            selectedCapture = capture
+            continue
+        }
+        
+        guard let newAppVersion = Double(newCapture.capturedOnAppVersion) else {
+            continue
+        }
+        
+        // Use the closest version possible to the current build version
+        if (newAppVersion > appVersion) {
+            selectedCapture = capture
+        }
+        
+    }
+    
+    return selectedCapture
+}
+
+internal func evaluatePageRules(screen: RowndScreen, rules: [MobileAppPageRuleUnknown]) -> Bool {
+    let result = rules.allSatisfy { rule in evaluatePageRule(rule: rule, screen: screen) }
+    return result
+}
+
+internal func evaluatePageRule(rule: MobileAppPageRuleUnknown, screen: RowndScreen) -> Bool {
+    switch rule {
+    case .unknown:
+        autoLogger.log("UNKNOWN RULE")
+        return false
+    case .or(let _rule):
+        return evaluatePageRule(rule: _rule, screen: screen)
+    case .and(let _rule):
+        return evaluatePageRule(rule: _rule, screen: screen)
+    case .rule(let _rule):
+        return evaluatePageRule(rule: _rule, screen: screen)
+    }
+}
+
+
+internal func evaluatePageRule(rule: MobileAppPageRule, screen: RowndScreen) -> Bool {
+    autoLogger.log("PAGE RULE \(rule)")
+    guard let operand = rule.operand else {
+        return false
+    }
+    
+    guard let operation = rule.operation else {
+        autoLogger.log("Operation is missing)")
+        return false
+    }
+    
+    if (operation == MobileAppPageRuleOperation.unknown) {
+        autoLogger.log("Operation type is unknown")
+        return false
+    }
+    
+    let value = rule.value
+    
+    do {
+        let jsonData = try JSONEncoder().encode(screen)
+        let query = jsonData.query(values: operand)?.first
+        guard let pageOperationsFunc = pageOperations[operation] else {
+            return false
+        }
+        let result = pageOperationsFunc(query, value)
+        return result
+    } catch {
+        autoLogger.log("Error encoding RowndScreen \(rule)")
+    }
+    
+    return false
+}
+
+internal func evaluatePageRule(rule: MobileAppPageAndRule, screen: RowndScreen) -> Bool {
+    let rules = rule.and
+    autoLogger.log("AND PAGE RULES \(rules)")
```

Comment:
Probably not super useful to keep these logs in permanently as i'm sure this would get rather noisy quite quickly

Reply:
Good point. Do you think we should add a toggle in the Rownd config to enable/disable specific logs like the autoLogger?

Reply:
Hmm I don't know where that should go. I'll have to think about it. Would be nice not to be able to flip a switch to turn on more verbose logging without needing to rebuild. But it's probably just going to be another developer troubleshooting anyway, so a config property in-app could work too. Long term, we're giong to want a strategy to debug automations on customers' users' devices as well.

Reply:
Log everything you think we need at the appropriate level (e.g., this should probably be a `debug` level log statement). Mac/iOS log filtering is quite excellent, so no need to enable/disable logs programmatically, as a log viewer can just show or hide whatever it needs to.